File: ForumWebProject.Application/Models/PostView.cs
namespace ForumWebProject.Application.Models;

public class PostView
{
    public Guid Id { get; set; }
    public int PostNumber { get; set; }
    
    public string? Text { get; set; }
    public DateTime DatePosted { get; set; }

    public Guid UserIdCreated { get; set; }
    public Guid TopicId { get; set; }

    public string? UserName { get; set; }
    public string? UserFirstName { get; set; }
    public string? UserLastName { get; set; }
    public DateTime UserRegistrationDate { get; set; }
}

File: ForumWebProject.Web/src/app/api/models/post-view.ts
/* tslint:disable */
/* eslint-disable */
export interface PostView {
  datePosted?: string;
  id?: string;
  postNumber?: number;
  text?: null | string;

  topicId?: string;
  userIdCreated?: string;
  replyToPostId?: undefined | string;

  userName?: string;
  userFirstName?: string;
  userLastName?: string;
  userRegistrationDate?:  any;

  // replyToUser?: null | string;
  // replyToText?: null | string;
  replyTo?: PostView | null;
}
